qiskit/algorithms/amplitude_estimators/ae.py

```diff
@@ -297,15 +333,37 @@ def estimate(self, estimation_problem: EstimationProblem) -> "AmplitudeEstimatio
         result.num_evaluation_qubits = self._m
         result.post_processing = estimation_problem.post_processing
 
-        if self._quantum_instance.is_statevector:
+        measurements = OrderedDict()
+        samples = OrderedDict()
+        if self._sampler is not None:
+            circuit = self.construct_circuit(estimation_problem, measurement=True)
+            job = self._sampler.run([circuit])
+            if job.status() is not JobStatus.DONE:
+                raise AlgorithmError("The job was not completed successfully. ")
+            ret = job.result()
+            result.shots = 1
+            circuit_results = {
+                np.binary_repr(k, circuit.num_qubits): v for k, v in ret.quasi_dists[0].items()
+            }
+            for state, probability in circuit_results.items():
```

Comment:
You mean reuse the function `evaluate_measurements` right ? I thought about something like this but this function is public and documented:
```
    def evaluate_measurements(
        self,
        circuit_results: Union[Dict[str, int], np.ndarray],
        threshold: float = 1e-6,
    ) -> Tuple[Dict[int, float], Dict[float, float]]:
        """Evaluate the results from the circuit simulation.

        Given the probabilities from statevector simulation of the QAE circuit, compute the
        probabilities that the measurements y/gridpoints a are the best estimate.

        Args:
            circuit_results: The circuit result from the QAE circuit. Can be either a counts dict
                or a statevector.
            threshold: Measurements with probabilities below the threshold are discarded.

        Returns:
            Dictionaries containing the a gridpoints with respective probabilities and
                y measurements with respective probabilities, in this order.
        """
```
That would mean altering its input from counts to probabilities with all the deprecation that would follow. I thought more direct to just copy part or the logic.

Reply:
Yeah I meant `evaluate_measurements` 👍🏻 

It would probably be nice to keep that function in future, for now we could also simply add support for `qiskit.result.QuasiDistribution` as a `circuit_results` and do the deprecation later (we'll anyways have to do that at some point I think) 🙂 

However in the interest of time we can also do this small refactor later and keep as is for now -- as you prefer! 🙂

Reply:
I added the quasi-distribution as a circuit_results for now.